core/src/main/java/org/elasticsearch/common/lucene/search/Queries.java

```diff
@@ -127,10 +127,13 @@ public static BooleanQuery applyMinimumShouldMatch(BooleanQuery query, @Nullable
             }
             builder.setMinimumNumberShouldMatch(msm);
             BooleanQuery bq = builder.build();
-            bq.setBoost(query.getBoost());
-            query = bq;
+            if (query.getBoost() != 1f) {
+                return new BoostQuery(bq, query.getBoost());
+            }
+            return bq;
+        } else {
```

Comment:
me too, but I find that this one is more readable this way...e.g. I forgot the second return above while changing this and introduced tons of failures :)